api/main_endpoints/util/Printer.js

```diff
@@ -0,0 +1,74 @@
+const fs = require('fs');
+const path = require('path');
+const logger = require('../../util/logger');
+const { MetricsHandler } = require('../../util/metrics.js');
+
+/**
+ * Deletes all chunks with the specified id from a directory
+ * @param {String} dir The directory the chunks are stored in
+ * @param {String} id  The UUID (36 char) assigned to the chunks
+ */
+async function cleanUpChunks(dir, id) {
+  try {
+    const chunks = await fs.promises.readdir(dir);
+
+    for (let chunk of chunks) {
+      if (!path.basename(chunk).includes(id)) continue;
+
+      await fs.promises.unlink(path.join(dir, chunk), err => {
+        logger.error(`Failed to delete chunk with id ${id} in ${dir}: ` + err);
+      });
+    }
+  } catch (err) {
+    logger.error('Encountered error while trying to delete chunks: ' + err);
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Deletes all chunks of specified age or older, 'expired', from a directory
+ * @param {String} dir The directory the chunks are stored in
+ * @param {Number} expiry Minimum age (ms) to be considered expired
+ */
+async function cleanUpExpiredChunks(dir, expiry) {
+  try {
+    const chunks = await fs.promises.readdir(dir);
+
+    for (let chunk of chunks) {
+      if (!['.pdf', '.CHUNK'].includes(path.extname(chunk))) continue;
+
+      const stats = await fs.promises.stat(path.join(dir, chunk));
+      const age = Date.now() - stats.mtimeMs;
+
+      if (age >= expiry) {
+        await fs.promises.unlink(path.join(dir, chunk), err => {
+          logger.warn(`Failed to delete expired chunk in ${dir}: ` + err);
+        });
```

Comment:
lets add 2 more metrics here. will help us see if cleanup is happening from incomplete requests
- number of expired chunks deleted
- total expired bytes deleted